New constructor for the SoAView and aggregate method for Collections (#47306)
cms-sw/cmssw: DataFormats/SoATemplate/README.md

Reviewer: Some strange has happened to this file.
Did you `clang-format` it ?

Author: Yes I did :(

`ConstViewTemplateFreeParams`. The parametrization of those templates is explained in the [Template
parameters section](#template-parameters).

> It is also possible to build a Generic View passing from the [Metarecords sublass](#metarecords-subclass). This

Reviewer: ```suggestion
It is also possible to build a generic `View` passing from the [Metarecords sublass](#metarecords-subclass). This
```

Reviewer: Actually, does it work for `ConstView` ?

> It is also possible to build a Customized View passing from the [Metarecords sublass](#metarecords-subclass). This

Author: We don't have many ideas about that. What about Scatter View?

> It is also possible to build a Customized View passing from the [Metarecords sublass](#metarecords-subclass). This

Reviewer: I'd suggest "disaggregated View" but that may be too complicated :-)
Maybe "generic View" ?

> view can give access to pointers not belonging to the same SoA and for this reason not contiguous in memory.

Reviewer: ```suggestion
view can pointe to data belonging to different SoAs and thus not contiguous in memory.
```

function computes the first byte of a structure right after a layout, allowing using a single buffer for multiple
layouts.

## Metarecords subclass

This subclass can be instanciated with the `records()` member function from a View and the member variables are
column objects. Every object contains the address to the first element of the column, the number of elements
per column and the stride for the Eigen columns. This is useful for building the Generic View described in
[View](#view) as soon as run-time check on the column sizes is performed. This nested class can be useful also
in the perspective of being able to manage columns of different sizes in the same data structure.  

> 

Reviewer: Can you mention the name of the class in the description ?
Something like
```
The nested type `Metarecords` describes the elements of the SoA. It can be instantiated by the `records()` member function from a `View` (or `ConstView` ?). Every object ...
```

Generic View -> generic `View`

What do you mean by "as soon as run-time check on the column sizes is performed" ?

I would drop the comment about "This nested class can be useful also...". You can add it later once the differently sized-columns are implemented.

Author: I mean that I am using these structs and not just addresses because I need to check the number of elements to be equal for every column.
